addons/website_slides/models/slide_channel.py

```diff
@@ -797,19 +797,38 @@ def toggle_active(self):
             to_activate.with_context(active_test=False).mapped('slide_ids').action_unarchive()
             super(Channel, to_activate).toggle_active()
 
-    @api.returns('mail.message', lambda value: value.id)
+    @api.returns("mail.message", lambda value: value.id)
     def message_post(self, *, parent_id=False, subtype_id=False, **kwargs):
-        """ Temporary workaround to avoid spam. If someone replies on a channel
+        """Make sure that only one review can be posted per course.
+        Also, temporary workaround to avoid spam. If someone replies on a channel
         through the 'Presentation Published' email, it should be considered as a
-        note as we don't want all channel followers to be notified of this answer. """
+        note as we don't want all channel followers to be notified of this answer."""
         self.ensure_one()
-        if kwargs.get('message_type') == 'comment' and not self.can_review:
-            raise AccessError(_('Not enough karma to review'))
+        if kwargs.get("subtype_xmlid") == "mail.mt_comment":
+            if self.env["mail.message"].search_count(
+                [
+                    ("res_id", "=", self.id),
+                    ("author_id", "=", self.env.user.partner_id.id),
+                    ("model", "=", "slide.channel"),
+                    ("subtype_id", "=", self.env.ref("mail.mt_comment").id),
+                ],
+                limit=1,
+            ):
+                raise ValidationError(_("Only a single review can be posted per course."))
+        if kwargs.get("message_type") == "comment" and not self.can_review:
+            raise AccessError(_("Not enough karma to review"))
```

Comment:
I would avoid "quotes" only diff on this method as it makes the new code much harder to review than it should in this case. Only adapt changes lines.